src/routes/login.tsx

```diff
@@ -111,17 +119,36 @@ export function LoginForm() {
               </div>
               <div className="grid gap-2">
                 {Role.Admin === role ? (
-                  <>
-                    <Label htmlFor="key" className="flex items-center">
+                  <div className="relative">
+                    <Label htmlFor="key" className="flex items-center mb-2">
                       Admin Key (xpriv)
                     </Label>
-                    <Input id="key" value={key} type="text" placeholder="Admin Key" onChange={onChangeKey} />
-                  </>
+                    <Input
+                      id="key"
+                      ref={inputRef}
+                      value={key}
+                      type={isPasswordVisible ? 'text' : 'password'}
+                      placeholder="Admin Key"
+                      onChange={onChangeKey}
+                      className="pr-12"
+                    />
+                    {isPasswordVisible ? (
+                      <EyeSlashIcon
+                        className="size-5 absolute top-8 right-3.5 cursor-pointer"
+                        onClick={handleTogglePasswordVisibility}
+                      />
+                    ) : (
+                      <EyeIcon
+                        className="size-5 absolute top-8 right-3.5 cursor-pointer"
+                        onClick={handleTogglePasswordVisibility}
+                      />
+                    )}
```

Comment:
Maybe instead of this copy/pasted code, we could do sth like this:

```ts
const ShowPasswordIcon = isPasswordVisible ? EyeSlashIcon : EyeIcon;
return (
    // (...)
    <ShowPasswordIcon
        className="size-5 absolute top-8 right-3.5 cursor-pointer"
        onClick={handleTogglePasswordVisibility}
     />
    //(...)
)
```

Reply:
Looks good. Will apply